pxf/pxf-hive/src/main/java/org/apache/hawq/pxf/plugins/hive/HiveORCBatchResolver.java

```diff
@@ -0,0 +1,257 @@
+import org.apache.hawq.pxf.api.io.DataType;
+import org.apache.hawq.pxf.api.utilities.ColumnDescriptor;
+import org.apache.hawq.pxf.api.utilities.InputData;
+import org.apache.hawq.pxf.api.utilities.Plugin;
+import org.apache.hawq.pxf.plugins.hive.utilities.HiveUtilities;
+import org.apache.hadoop.hive.serde2.*;
+import org.apache.hadoop.hive.serde2.objectinspector.*;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.*;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector.Category;
+import org.apache.hadoop.hive.serde2.objectinspector.PrimitiveObjectInspector.PrimitiveCategory;
+import org.apache.hadoop.hive.ql.exec.vector.*;
+
+@SuppressWarnings("deprecation")
+public class HiveORCBatchResolver extends Plugin implements ReadVectorizedResolver {
+
+    private static final Log LOG = LogFactory.getLog(HiveORCBatchResolver.class);
+
+    private List<List<OneField>> resolvedBatch;
+    private StructObjectInspector soi;
+
+    public HiveORCBatchResolver(InputData input) throws Exception {
+        super(input);
+        try {
+            soi = (StructObjectInspector) HiveUtilities.getOrcReader(input).getObjectInspector();
+        } catch (Exception e) {
+            LOG.error("Unable to create an object inspector.");
+            throw e;
+        }
+    }
+
+    @Override
+    public List<List<OneField>> getFieldsForBatch(OneRow batch) {
+
+        Writable writableObject = null;
+        Object fieldValue = null;
+        VectorizedRowBatch vectorizedBatch = (VectorizedRowBatch) batch.getData();
+
+        // Allocate empty result set
+        resolvedBatch = new ArrayList<List<OneField>>(vectorizedBatch.size);
+        for (int i = 0; i < vectorizedBatch.size; i++) {
+            ArrayList<OneField> row = new ArrayList<OneField>(inputData.getColumns());
+            resolvedBatch.add(row);
+            for (int j = 0; j < inputData.getColumns(); j++) {
+                row.add(null);
+            }
+        }
+
+        /* process all columns*/
+        for (int columnIndex = 0; columnIndex < vectorizedBatch.numCols; columnIndex++) {
+            ObjectInspector oi = soi.getAllStructFieldRefs().get(columnIndex).getFieldObjectInspector();
+            if (oi.getCategory() == Category.PRIMITIVE) {
+                PrimitiveObjectInspector poi = (PrimitiveObjectInspector ) oi;
+                resolvePrimitiveColumn(columnIndex, oi, vectorizedBatch);
+            } else {
+                throw new UnsupportedTypeException("Unable to resolve column index:" +  columnIndex + ". Only primitive types are supported.");
+            }
+        }
+
+        return resolvedBatch;
+    }
+
+    private void resolvePrimitiveColumn(int columnIndex, ObjectInspector oi, VectorizedRowBatch vectorizedBatch) {
+
+        OneField field = null;
+        Writable writableObject = null;
+        /* process all rows from current batch for given column */
+        for (int rowIndex = 0; rowIndex < vectorizedBatch.size; rowIndex++) {
```

Comment:
Since we do know that the whole all the writableObjects are of the same type, why don't we simply extract all values in the column into a Writable [] and resolve the entire writable array at once using just one switch case to determine the field type and resolve each writable value in one go ?